packages/next/client/image.tsx

```diff
@@ -102,11 +102,10 @@ type CallLoaderProps = {
 type URLResolver = (resolverProps: CallLoaderProps) => string
 
 let customResolver: URLResolver
+let resolverError = false
 export function registerCustomResolver(resolver: URLResolver): void {
-  if (process.env.NODE_ENV !== 'production' && configLoader !== 'custom') {
-    throw new Error(
-      `registerCustomResolver can only be used if image loader is set to 'custom' in next.config.js`
-    )
+  if (configLoader !== 'custom') {
+    resolverError = true
```

Comment:
What's the reason to delay the error instead of throw here?

Reply:
I noticed when writing the tests that if it throws during _app.js script evaluation, the page renders an internal server error instead of the red box.